### PR TITLE
feat: support cls-hooked / tests

### PR DESCRIPTION
**Pull request by sushantdhiman**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [ ] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [ ] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Have you added new tests to prevent regressions?
- [x] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Did you update the typescript typings accordingly (if applicable)?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/master/CONTRIBUTING.md)?

### CODE REVIEW

File: lib/sequelize.js [hunk 1]
```diff
@@ -1130,8 +1129,9 @@ class Sequelize {
     // save namespace as `Sequelize._cls`
     this._cls = ns;
 
-    // patch bluebird to bind all promise callbacks to CLS namespace
-    clsBluebird(ns, Promise);
+    Promise.config({
+      asyncHooks: true
```
Comment (jedwards1211) [inline lib/sequelize.js:1133]: @sushantdhiman builtlin Promises automatically support async hooks right?  Want to know if I need to replace this with something when we get rid of bluebird.

Comment (papb) [reply lib/sequelize.js:1133]: Good question @jedwards1211, I don't know the answer, hopefully @sushantdhiman does. However, I'd guess that you could just try removing it, in theory we should have tests that fail if this stops working...

Comment (sushantdhiman) [reply lib/sequelize.js:1133]: @jedwards1211 No, you need to manually enable them, https://nodejs.com/api/async_hooks.html#async_hooks_promise_execution_tracking. But we don't need to enable anything. `cls-hooked` enable those api when namespace is created.

So no change will be required at sequelize level

Comment (jedwards1211) [reply lib/sequelize.js:1133]: okay great

**Pull request merged**